### ISSUE TITLE
renamecols function in transform

### ISSUE THREAD

**Issue by jariji** [JuliaData/DataFrames.jl#3122]
`innerjoin` takes a `renamecols` function but `transform` doesn't. I want to say `transform(... => AsTable; renamecols=x->"foo$x")`.

**Comment by bkamins** [comment]
Let us keep it for a decision. In an initial design it was decided that adding this would lead to too much complexity of the minilanguage.

**Comment by jariji** [comment]
Use case: I had a table-valued column and I wanted to split it into top-level columns but keep some reference to where they came from, naming them by their originating column.

**Comment by bkamins** [comment]
Yes - I understand the use-case. The point here is only the complexity of the operation specification language. There is a mental limit that users can digest the complexity of it and it is already quite complex.

We will have to think if the use-case is frequent enough to warrant adding additional level of complexity. The point is, that currently, although it is not super convenient for your use case you can write:

```
... => vector_of_column_names
```
so you can achieve what you want, with the only limitation that you have to do this step of giving output column names manually.
